PORTING.md

```diff
@@ -0,0 +1,354 @@
+# Porting to xpcc
+
+## What xpcc supports
+
+xpcc contains extensive support for AVR and STM32 devices, specifically:
+
+- AT90, ATtiny, ATmega devices
+- STM32 F0, F1, F3, F4, F7 devices.
+
+If your device falls into any of those, you can just use the specific device
+identifier in your `project.cfg` and it'll very likely just work out-of-the-box.
+
+If you want to add a development board for an existing device, have a look at [`xpcc/architecture/platform/board`](https://github.com/roboterclubaachen/xpcc/tree/develop/src/xpcc/architecture/platform/board).
+
+## Porting a new device from AVR or STM32
+
+It may be as easy as adding a new device file, but if it's a brand new device, the device headers also need to be updated. When porting a new series (for example the STM32L4), a lot of code can be reused, but there may be very important differences between the two that are not immediately obvious to someone who isn't that familiar with STM32 devices.
+
+We recommend writing us an email, or opening up an issue on GitHub.
+
+## Porting a new device from a new vendor
+
+Note: We assume this new device is Cortex-M based. For other CPU architectures, send us an email.
+
+Porting to a new vendor requires you to:
+
+1. Add a bunch of header files for the new device.
+2. Add a device files for the new device.
+3. Make the build system aware of the new device classes.
+4. Adapt the Cortex-M startup- and linkerscripts.
+5. Compile a minimal example.
+6. Write a basic GPIO and delay driver.
+7. Compile an LED blinking example.
```

Comment:
I'll just make it clearer in the text that this is an overview. I'm too old to battle Markdown. #LifeCrisis